modules/ROOT/pages/datasense-perceptive-flow-design-concept.adoc

```diff
@@ -6,45 +6,60 @@ endif::[]
+== Specifically Declaring Metadata
 
-=== Being More Specific About Your Metadata Declarations
+All components in a Mule application let you manually define the metadata that they expose. When building prescriptive and reusable components in a flow, you need to pay attention when to let Datasense handle the metadata that a component expects, and when you need to manually define a metadata that a component should expect or expose.
 
-Using the transform message component's output metadata as expected metadata could be wrong. By explicitly declaring a metadata type, you are masking the actual metadata type of the data coming out from your transform message component. DataWeave transform message implementation can completely infer this output data type. +
-The best practice is to define expected metadata where this data is used or consumed, and it is not automatically inferred (static/dynamic metadata).
+We recommend you to specifically declare metadata only when:
 
-At the same time, defining the input metadata for your Transform Message component also masks the actual incoming metadata coming into your transform message component to be used on mapping. +
-It would be best to define metadata where data is produced, and again, only when it is not automatically inferred (static/dynamic metadata). For example, in the source or the extension operation that generated the actual message payload.
+* It is absolutely required, for example, metadata for a value being defined by you.
+* When it is most cohesive to an annotated value, for example, when metadata for a value is being used or expected by a component.
 
-With this concept in mind, always remember that whenever data lives outside the scope of a reusable artifact, you should also declare it as part of that artifact's contract. +
-Flows and subflows, for example, are expected to declare their metadata contract, just like any other processor.
+For example, the transform message component should not define a metadata contract because Dataweave acts as a data mediation device between the producer and consumer of data, Those boundaries are the ones that should define the contracts. By explicitly declaring a metadata type, you are masking the actual metadata type of the data coming out from your transform message component. DataWeave transform message implementation can completely infer this output data type. The best practice is to define expected metadata where this data is used or consumed, and it is not automatically inferred. At the same time, defining the input metadata for your Transform Message component also masks the actual incoming metadata coming into your transform message component to be used on mapping. It would be best to define metadata where data is produced, and again, only when it is not automatically inferred.
```

Comment:
Transform Message shows capitalized in some instances and others don't, please check for consistency.